ResearchKit/Common/ORKWebViewStepViewController.m

```diff
@@ -55,6 +55,11 @@ - (void)stepDidChange {
     
     if (self.step && [self isViewLoaded]) {
         WKWebViewConfiguration *config = [[WKWebViewConfiguration alloc] init];
+        config.allowsInlineMediaPlayback = true;
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        config.mediaPlaybackRequiresUserAction = false;
```

Comment:
Can we avoid using a deprecated API?

Reply:
Were you thinking something like the below, which would work on iOS 8+?

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
        if ([config respondsToSelector:@selector(mediaTypesRequiringUserActionForPlayback)]) {
            config.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeNone;
        } else if ([config respondsToSelector:@selector(requiresUserActionForMediaPlayback)]) {
            config.requiresUserActionForMediaPlayback = false;
        } else {
            config.mediaPlaybackRequiresUserAction = false;
        }
#pragma clang diagnostic pop
```

Or to just use the latest and support iOS 10+?

```
        if ([config respondsToSelector:@selector(mediaTypesRequiringUserActionForPlayback)]) {
            config.mediaTypesRequiringUserActionForPlayback = WKAudiovisualMediaTypeNone;
        }
```

Reply:
I would say use the latest since ResearchKit 2.0 sets the base SDK to iOS 11.0.

Reply:
Fixed